example/shell.nix

```diff
@@ -0,0 +1,33 @@
+{ nixpkgs ? <nixpkgs> }:
+let
+  pkgs = import (builtins.fetchTarball {
```

Comment:
Why not do this at the system level? If they're just doing it for one thing we can install HIE globally.

Reply:
Also, why do we need to pin this? A lot of things here are using the ambient nixpkgs, either we should just assume that it's got a working vscode or we should pin it at the top level.

Reply:
pinning here means we can use it on any computer and it will work, for example on my mac it works. Same with hie, it all just works having it in this nix shell

Reply:
For this example I just prefer a known-good setup that should work for everyone

Reply:
I guess my concern is that e.g. we're not getting a pinned version of the nixos virtualbox bits, but maybe that's okay.

Reply:
we are not using anything unstable in the nixos config though and they're pretty good about keeping that working well. There's almost nothing that depends on `<nixpkgs>` in there anyway.